pages/assets/[id].tsx

```diff
@@ -49,14 +49,19 @@ const Asset: React.FC<Props> = () => {
 
  if (asset && asset.status === 'errored') {
     const message = asset.errors && asset.errors.messages[0];
-    errorMessage = `Error creating this asset: ${message}`;
-  }
+    errorMessage = {
+      initial: 'Error creating this asset: Please upload a valid video file (gifs are not supported)',
+      apiError: message
+  }}
 
   if (errorMessage) {
     return (
       <Layout darkMode={false}>
-        <div><h1>{errorMessage}</h1></div>
+        <div><h1>{errorMessage.initial}</h1></div>
```

Comment:
There's a view other places where set set `errorMessage =` to a string (line 28 and line 32). Honestly I would expect the typescript compiler to catch this. I'm wondering why it didn't.

I suggest

`errorMessage = ` is always set to a string and it's shown in the `<h1>`
`errorDetails=` is a different string that only gets set when the asset is errored. And then do something like this:

```tsx
{errorDetails && <p>{errorDetails}</p>}
```

lmk what you think of that

Reply:
yea makes sense!